Push API (#290)
WalletConnect/walletconnect-docs: docs/specs/push/push-proposal.md

# Push Proposal

## User Flow

> User vists application that connected with its wallet

Reviewer: Typo:
```suggestion
User visits application that connected with its wallet
```

> App can prompt a push proposal after connecting or it can provide a "bell" button that user can trigger the push proposal

Reviewer: ```suggestion
App can prompt a push proposal after connecting or it can provide a "bell" button for the user to trigger the push proposal
```

> User can approve or reject the push proposal from its wallet.

Reviewer: ```suggestion
User can approve or reject the push proposal from their wallet.
```

> In future events, the app can trigger push notitifcations after subscriptions has been approved.

Reviewer: Fixing typo:

```suggestion
In future events, the app can trigger push notifications after subscription has been approved.
```

## Proposal Protocol

#### Prerequisites
Wallet and Dapp are required to establish pairing P before proceeding to Push protocol execution.

#### Protocol

Proposal protocol will be established as follows:

> 1. Dapp sends push proposal on pairing P with publicKey, relay and metadata

Reviewer 2: ```suggestion
1. Dapp sends push proposal on pairing P with publicKey X, relay and metadata
```

> 2. Wallet receives push proposal with public key X on pairing P

Reviewer 2: ```suggestion
2. Wallet receives push proposal with publicKey X on pairing P
```

3. Wallet generates key pair Y

> 4. Wallet derivees symmetric key derived with keys X and Y

Reviewer 2: ```suggestion
4. Wallet derives symmetric key derived with keys X and Y
```
